vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -390,24 +461,47 @@ public void createIndex() throws JsonProcessingException {
 				.bodyToMono(Void.class)
 				.block();
 		}
-		catch (WebClientResponseException exception) {
-			logger.warn("Index already exists");
-		}
 		catch (Exception e) {
-			throw new RuntimeException("Internal server error", e);
+			logger.warn("An unexpected error occurred while creating the index");
 		}
 	}
 
 	public void deleteIndex() {
 		try {
-			client.delete().uri("/" + INDEX_NAME + "/?delete-data=true").retrieve().bodyToMono(Void.class).block();
-		}
-		catch (WebClientResponseException.NotFound notFoundException) {
-			logger.warn("Vector index does not exist", notFoundException);
+			DeleteRequest deleteRequest = new DeleteRequest();
+			deleteRequest.setDeleteData(true);
+			client.method(HttpMethod.DELETE)
+				.uri("/" + INDEX_NAME)
+				.body(BodyInserters.fromValue(deleteRequest))
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
 		}
 		catch (Exception e) {
-			throw new RuntimeException("An unexpected error occurred", e);
+			logger.warn("An unexpected error occurred while deleting the index", e);
```

Comment:
Should this throw an exception or just log?  

I think these cases where it's a void return - we might want to throw an exception to let the caller know something didn't work.  

In the cases where there is a boolean return type or any status return type, we could let the caller know through the return value?